### PR TITLE
build: allow bazel build ...

### PR DESCRIPTION
**Pull request by alexeagle**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by alexeagle** [comment]
Note, the reason this PR removes `firebase-tools` is:

1) firebase-tools has an optional dependency on https://www.npmjs.com/package/@google-cloud/functions-emulator
2) yarn's `--ignore-optional` doesn't work for transitive deps, so there's no way to yarn install without getting that functions-emulator package
3) functions-emulator has a transitive dep on `grpc`
4) the version of `grpc` we get has `BUILD` files and no `WORKSPACE` file so it always breaks `bazel build ...`

It could be solved by any of:
1) remove firebase-tools - this is what I did
2) fix yarn so you can omit optional deps of a transitive dep
3) make functions-emulator depend transitively on a more recent `grpc` version
4) patch `grpc` after install by doing an `rm` command in our postinstall or something

**Comment by gkalpak** [comment]
Maybe one could use [resolutions][1] to pin `grpc` to a new version (as long as it is compatible with `functions-emulator` - or maybe we don't care :grin:)

[1]: https://yarnpkg.com/en/docs/package-json#toc-resolutions

**Comment by alexeagle** [comment]
Thanks @gkalpak I'll try that if Igor doesn't like this approach

File: package.json [hunk 1]
```diff
@@ -63,7 +63,6 @@
     "cors": "2.8.4",
     "domino": "1.0.29",
     "entities": "1.1.1",
-    "firebase-tools": "3.12.0",
```
Comment (IgorMinar) [inline package.json]: this change has nothing to do with the `bazel build ...` commit. Please move to a separate commit, or remove in favor of #22186

**Review by IgorMinar — APPROVED** [review]
the rest looks great. thanks

File: package.json (hunk 1 above)
Comment (alexeagle) [reply package.json]: like I explained above, this is *required* for `bazel build ...` - it's the main thing that was in the way

**Review by IgorMinar — APPROVED** [review]
please amend the commit message with explanation about the protobuffjs

File: package.json (hunk 1 above)
Comment (alexeagle) [reply package.json]: copied the explanation to the commit message

**Comment by vicb** [comment]
This PR caused master to go red:
- firebase is required to publish the build artifacts size ATM,
- bumping zone.js seems to be a pb (needs to be investiagted)

**Review by mhevery — APPROVED** [review]

**Comment by matsko** [comment]
@alexeagle please rebase onto master.

**Comment by alexeagle** [comment]
Rebased

**Review by IgorMinar — APPROVED** [review]
